### PR TITLE
mission-planner: init at version 1.3.74

### PR DESCRIPTION
**Pull request by wucke13**
###### Motivation for this change
Configuring my Ardupilot FC

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wucke13** [comment]
@GrahamcOfBorg build mission-planner

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  nativeBuildInputs = [ makeWrapper mono6 unzip ];
+  sourceRoot = ".";
+
+  AOT_FILES = [ "MissionPlanner.exe" "MissionPlanner.*.dll" ];
+
+  buildPhase = ''
+    for file in $AOT_FILES
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:29]:
```suggestion
    for file in $AOT_FILES; do
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      mono --aot $file
+    done
+  '';
+
+  installPhase = ''
+    # create target directoryies
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:35]:
```suggestion
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    mkdir -p $out/{bin,opt/mission-planner}
+    
+    # install icon & desktopitem
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:38]:
```suggestion
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    install -m 444 -D mpdesktop150.png $out/share/icons/mission-planner.png
+    cp -r ${desktopItem}/share/applications $out/share/
+
+    # install MissionPlanner
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:42]:
```suggestion
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    mv * $out/opt/mission-planner
+
+    # create mono wrapper for MissionPlanner
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:45]:
```suggestion
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  buildPhase = ''
+    for file in $AOT_FILES
+    do
+      mono --aot $file
+    done
+  '';
+
+  installPhase = ''
+    # create target directoryies
+    mkdir -p $out/{bin,opt/mission-planner}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:36]:
```suggestion
    mkdir -p $out/bin $out/opt/mission-planner
```

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  meta = with stdenv.lib; {
+    description = "An ArduPilot ground station";
+    longDescription = ''
+      Full-featured ground station application for the ArduPilot open source
+      autopilot project.  Lets you both flash, configure and control ArduPilot
+      Plane, Copter and Rover targets.
+    '';
+    homepage = http://www.gnu.org/software/hello/manual/;
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.eelco ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/robotics/mission-planner/default.nix:59]: Please do not add other people as maintainer.

Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:59]: That was a mistake, fixed it!

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 6 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:36]: Why is more repetition better?

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 2 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: Why are comments bad?

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:29]: What is the rational here? What makes one better than the other in you point of view?

Comment (SuperSandro2000) [reply pkgs/applications/science/robotics/mission-planner/default.nix:29]: It uses one line less and is just a common thing I do.

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: Comments for complex intuitive parts of code are good. On the other hand comments for easy to overlook and understand code are not that useful and easily get out of date. Creating directories with ``mkdir -p`` is a very common thing in nix and does not require a comment.

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/robotics/mission-planner/default.nix:36]: Using ``{..,..}`` is nice if only a small part of a path changes like the last directory in a 5 deep nested path. I don't think it makes the code more readable if ``$out/`` is the only repeating part.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107545` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mission-planner</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
There's an editorconfig issue also.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107545` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mission-planner</li>
  </ul>
</details>

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:29]: I prefer GNU C style, and similar to that, I think having the loop bodies opening symbol in the next line makes the code more readable.

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 2 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: Ok then

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 6 above)
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:36]: I disagree on this one as well, but it likely boils down to personal preference. Is it a pressing issue to you having this changed?

**Comment by wucke13** [comment]
@SuperSandro2000 @doronbehar Thank you two for the feedback so far, I adjusted some of the issues you mentioned. Hope it works out now!

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    description = "An ArduPilot ground station";
+    longDescription = ''
+      Full-featured ground station application for the ArduPilot open source
+      autopilot project.  Lets you both flash, configure and control ArduPilot
+      Plane, Copter and Rover targets.
+    '';
+    homepage = http://www.gnu.org/software/hello/manual/;
```
Comment (doronbehar) [inline pkgs/applications/science/robotics/mission-planner/default.nix:50]:
```suggestion
    homepage = "https://ardupilot.org/planner/";
```

Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:50]: Oh no, that's a nifty one :D

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.wucke13 ];
```
[the remark this answers is not in the document]
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:52]: Actually, I'd only do this if there'd be multiple maintainers. Are you in? :smile:

File: pkgs/applications/science/robotics/mission-planner/default.nix [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+  buildPhase = ''
+    for file in $AOT_FILES
+    do
+      mono --aot $file
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{bin,opt/mission-planner}
```
[the remark this answers is not in the document]
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: What would you like to see here? Shall I move my custom phases to become `pre` phases? Other than that, I can't see how any of changes in the docs could be used for this?

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/science/robotics/mission-planner/default.nix:52]: This makes it easier adding new maintainers and a style which is preferred by me, too.

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 10 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: He is talking about hooks like https://github.com/NixOS/nixpkgs/blob/e7fe577d9803885d1191c6612b95c246cb605dde/pkgs/build-support/ocaml/oasis.nix#L24. Some other hooks add things to those phases which could be skipped.

[the remark this answers is not in the document]
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix:35]: Ah, thank you! Adjusted accordingly.

File: pkgs/applications/science/robotics/mission-planner/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (wucke13) [reply pkgs/applications/science/robotics/mission-planner/default.nix]: Well then

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107545` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mission-planner</li>
  </ul>
</details>

**Pull request merged**